osu.Game/Database/LegacyExporter.cs

```diff
@@ -33,17 +33,17 @@ protected LegacyExporter(Storage storage)
             UserFileStorage = storage.GetStorageForDirectory(@"files");
         }
 
+        protected virtual string GetItemExportString(TModel item) => item.GetDisplayString().GetValidFilename();
```

Comment:
I would probably just call this `GetFilename`. Also any operation that's performed on all overrides of the method should be moved out when possible (i.e. `GetValidFilename` should be called in `Export` rather than on every override of the method).